### PR TITLE
[Docs] Add AVR and ARM examples to GPIO Commands

### PR DESCRIPTION
**Pull request by drashna**
Add examples for reference for people not as well versed in microcontroller coding, such as myself.

## Types of Changes

- [X] Documentation

## Issues Fixed or Closed by This PR

* My frequent annoyance, since I have issues remembering this stuff

### CODE REVIEW

File: docs/internals_gpio_control.md [hunk 1]
```diff
@@ -6,17 +6,16 @@ QMK has a GPIO control abstraction layer which is microcontroller agnostic. This
...
-|`setPinInputLow(pin)` |Set pin as input with build in pull-down (Supported only on STM32)|
-|`setPinOutput(pin)`   |Set pin as output                                                 |
-|`writePinHigh(pin)`   |Set pin level as high, assuming it is an output                   |
-|`writePinLow(pin)`    |Set pin level as low, assuming it is an output                    |
-|`writePin(pin, level)`|Set pin level, assuming it is an output                           |
-|`readPin(pin)`        |Returns the level of the pin                                      |
-
+|Function              |Description                                                       | Old AVR Examples                               | Old ChibiOS/ARM Examples                      |
+|----------------------|------------------------------------------------------------------|------------------------------------------------|-----------------------------------------------|
+|`setPinInput(pin)`    |Set pin as input with high impedance (High-Z)                     | DDRB &= ~(1<<2)                                | palSetLineMode(pin, PAL_MODE_INPUT)           |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]: Needs code tags, the pipe character messes up the table formatting :)

Comment (drashna) [reply docs/internals_gpio_control.md]: lol, whoops, will fix.

**Comment by drashna** [comment]
@fauxpark okay, should be fixed now, though can't guarantee that it will work in the doc site, because ... weird rendering issues. :D

**Comment by fauxpark** [comment]
You should be able to preview it locally by `cd`ing to `docs` and running `python -m http.server 3000` 😃

File: docs/internals_gpio_control.md [hunk 2]
```diff
@@ -6,18 +6,16 @@ QMK has a GPIO control abstraction layer which is microcontroller agnostic. This
...
+|`setPinInputLow(pin)` |Set pin as input with build in pull-down (Supported only on STM32)| N/A (Not supported on AVR)                     | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)`  |
+|`setPinOutput(pin)`   |Set pin as output                                                 | `DDRB \|= (1<<2)`                               | `palSetLineMode(pin, PAL_MODE_OUTPUT_PUSHPULL)` |
+|`writePinHigh(pin)`   |Set pin level as high, assuming it is an output                   | `PORTB \|= (1<<2)`                              | `palSetLine(pin)`                               |
+|`writePinLow(pin)`    |Set pin level as low, assuming it is an output                    | `PORTB &= ~(1<<2)`                             | `palClearLine(pin)`                             |
+|`writePin(pin, level)`|Set pin level, assuming it is an output                           | `(level) ? PORTB \|= (1<<2) : PORTB &= ~(1<<2)` | `(level) ? palSetLine(pin) : palClearLine(pin)` |
+|`readPin(pin)`        |Returns the level of the pin                                      | `_SFR_IO8(pin >> 4) & _BV(pin & 0xF)`          | `palReadLine(pin)`                              |
 ## Advanced Settings
```
Comment (noroadsleft) [inline docs/internals_gpio_control.md]: Need a line break before this heading.

image

```suggestion

## Advanced Settings
```

File: docs/internals_gpio_control.md [hunk 3]
```diff
@@ -6,18 +6,16 @@ QMK has a GPIO control abstraction layer which is microcontroller agnostic. This
...
-|`writePinHigh(pin)`   |Set pin level as high, assuming it is an output                   |
-|`writePinLow(pin)`    |Set pin level as low, assuming it is an output                    |
-|`writePin(pin, level)`|Set pin level, assuming it is an output                           |
-|`readPin(pin)`        |Returns the level of the pin                                      |
-
+|Function              |Description                                                       | Old AVR Examples                               | Old ChibiOS/ARM Examples                        |
+|----------------------|------------------------------------------------------------------|------------------------------------------------|-------------------------------------------------|
+|`setPinInput(pin)`    |Set pin as input with high impedance (High-Z)                     | `DDRB &= ~(1<<2)`                              | `palSetLineMode(pin, PAL_MODE_INPUT)`           |
+|`setPinInputHigh(pin)`|Set pin as input with build in pull-up                            | `DDRB &= ~(1<<2); PORTB \|= (1<<2)`             | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)`    |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
|`setPinInputHigh(pin)`|Set pin as input with builtin pull-up resistor                    | `DDRB &= ~(1<<2); PORTB \|= (1<<2)`             | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)`    |
```

File: docs/internals_gpio_control.md [hunk 4]
```diff
@@ -6,18 +6,16 @@ QMK has a GPIO control abstraction layer which is microcontroller agnostic. This
...
+|`setPinInputLow(pin)` |Set pin as input with build in pull-down (Supported only on STM32)| N/A (Not supported on AVR)                     | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)`  |
```
Comment (fauxpark) [inline docs/internals_gpio_control.md]:
```suggestion
|`setPinInputLow(pin)` |Set pin as input with builtin pull-down resistor                  | N/A (Not supported on AVR)                     | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)`  |
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**